firebase-firestore/src/main/java/com/google/firebase/firestore/util/AsyncQueue.java

```diff
@@ -113,7 +116,6 @@ void skipDelay() {
      * the AsyncQueue) provides a guarantee that the task will not be run.
      */
     public void cancel() {
-      verifyIsCurrentThread();
```

Comment:
Hrm...  This assert is valid and serves an important purpose (ensures that we can provide the guarantee described in the method comment).  So I am not wild about removing it rather than figuring out a way to fix the tests (If we make SqlitePersistence AsyncQueue-aware then we really should be calling all of its methods on the AsyncQueue).

Worst case if we want a cheap fix maybe we can try/catch the exception from tests?  Else we need some sort of better plumbing around PersistenceTestHelpers so we can recover the AsyncQueue or something...